molgenis-api-permissions/src/main/java/org/molgenis/api/permissions/model/response/LabelledPermissionResponse.java

```diff
@@ -0,0 +1,49 @@
+package org.molgenis.api.permissions.model.response;
+
+import com.google.auto.value.AutoValue;
+import java.util.Set;
+import javax.annotation.Nullable;
+import org.molgenis.util.AutoGson;
+
+@AutoValue
+@AutoGson(autoValueClass = AutoValue_LabelledPermissionResponse.class)
+@SuppressWarnings(
+    "squid:S1610") // Abstract classes without fields should be converted to interfaces
+public abstract class LabelledPermissionResponse {
+  @Nullable
+  public abstract String getUser();
+
+  @Nullable
+  public abstract String getRole();
+
+  @Nullable
+  public abstract String getObjectId();
+
+  @Nullable
+  public abstract String getLabel();
+
+  @Nullable
+  public abstract String getTypeLabel();
+
+  @Nullable
+  public abstract String getTypeId();
+
+  @Nullable
+  public abstract String getPermission();
+
+  @Nullable
+  public abstract Set<LabelledPermissionResponse> getInheritedPermissions();
```

Comment:
empty set instead of nullable?

Reply:
either add a doc 'can be null in case inherited permissions are not requested' or use extension

Reply:
doc added, commented line not changed, so not marked as outdated by github